pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -260,22 +228,19 @@ public SendMessageInfo sendMessages(final List<Entry> entries, SendListener list
                             consumerId, pos.getLedgerId(), pos.getEntryId());
                 }
 
-                // We only want to pass the "real" promise on the last entry written
-                ChannelPromise promise = ctx.voidPromise();
-                if (i == (entries.size() - 1)) {
-                    promise = writePromise;
-                }
-                int redeliveryCount = subscription.getDispatcher().getRedeliveryTracker().getRedeliveryCount(PositionImpl.get(messageId.getLedgerId(), messageId.getEntryId()));
-                ctx.write(Commands.newMessage(consumerId, messageId, redeliveryCount, metadataAndPayload), promise);
+                int redeliveryCount = subscription.getDispatcher().getRedeliveryTracker()
```

Comment:
A nit, but the redeliveryTracker could be passed as an argument, or even on construction of the dispatcher to avoid reaching into the dispatcher.

Reply:
Fixed